pkg/block/local/adapter.go

```diff
@@ -79,18 +81,33 @@ func resolveNamespace(obj block.ObjectPointer) (block.QualifiedKey, error) {
 	return qualifiedKey, nil
 }
 
+// verifyPath ensures that p is under the directory controlled by this adapter.  It does not
+// examine the filesystem and can mistakenly error out when symbolic links are involved.
+func (l *Adapter) verifyPath(p string) error {
```

Comment:
This new check breaks existing local repos, no?

Reply:
I can see it breaking an existing repo *if*:
1. It uses the local block adapter.  (See #1998 for some more work against anyone doing this for real...).
2. One of the "create from staging" (direct upload) APIs was used to access files *not* under the root of local storage.

I *could* add a startup check, it would go over the entire DB and see that everything is OK.  Not sure it is warranted.  @ozkatz do you have a producty opinion?